pyccel/codegen/printing/ccode.py

```diff
@@ -1186,6 +1216,11 @@ def formatted_args_to_printf(args_format, args, end):
                                                 file=expr.file)],
                                  unravelled = True)
                 code += self._print(body)
+            elif isinstance(f, LiteralString):
+                arg_format, arg = self.get_print_format_and_arg(f)
+                if isinstance(arg, str):
+                    arg = self._undo_print_LiteralString(arg)
+                    args_format.append(arg[1:-1])
```

Comment:
This solution seems overly complicated to me. You add the escape sequences in `get_print_format_and_arg` and revert then immediately after. Would it not be simpler to work with the string saved in the `LiteralString` object directly? (`f.python_value`)